src/app/retrospective/sprint-detail/sprint-detail.component.ts

```diff
@@ -93,11 +95,19 @@ export class SprintDetailComponent implements OnInit, OnDestroy  {
         return (featureType + taskType + bugType);
     }
 
+    refreshSprintDetails() {
+        this.toggleToTriggerRefresh = !this.toggleToTriggerRefresh;
+        this.sprintDetailsRefreshComplete = false;
+        const sprintDetailsObservable = this.getSprintDetails(true);
+        sprintDetailsObservable.subscribe(() => {}, () => {}, () => {
+                this.sprintDetailsRefreshComplete = true;
+            }
+        );
+    }
+
     getSprintDetails(isRefresh = false) {
-        if (isRefresh) {
-            this.toggleToTriggerRefresh = !this.toggleToTriggerRefresh;
-        }
-        this.retrospectiveService.getSprintDetails(this.retrospectiveID, this.sprintID).subscribe(
+        const sprintDetailsObservable = this.retrospectiveService.getSprintDetails(this.retrospectiveID, this.sprintID);
+        sprintDetailsObservable.subscribe(
```

Comment:
Instead of subscribing to this Observable twice, can we use the `do() operator` to perform the subsequent actions? As I can see from [the docs](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-do),
> Perform a side effect for every emission on the source Observable, but return an Observable that is identical to the source.